src/config.rs

```diff
@@ -694,17 +694,17 @@ impl<'a> Cfg<'a> {
     }
 
     #[tracing::instrument(level = "trace")]
-    pub(crate) async fn active_rustc_version(&mut self) -> Result<String> {
+    pub(crate) fn active_rustc_version(&mut self) -> Result<Option<String>> {
         if let Some(t) = self.process.args().find(|x| x.starts_with('+')) {
             trace!("Fetching rustc version from toolchain `{}`", t);
             self.set_toolchain_override(&ResolvableToolchainName::try_from(&t[1..])?);
         }
 
         Ok(self
-            .find_or_install_active_toolchain()
-            .await?
-            .0
-            .rustc_version())
+            .find_active_toolchain()?
```

Comment:
How do you feel about this alternative?

```rust
-        Ok(self
-            .find_active_toolchain()?
-            .map(|(name, _)| Toolchain::new(self, name))
-            .transpose()?
-            .map(|toolchain| toolchain.rustc_version()))
+        Ok(match self.find_active_toolchain()? {
+            Some((name, _)) => Some(Toolchain::new(self, name)?.rustc_version()),
+            None => None,
+        })
```

Reply:
@djc Now that I think about it, I guess let-else would be a better alternative.

Reply:
@djc Changed to

```rust
let Some((name, _)) = self.find_active_toolchain()? else {
    return Ok(None);
};
Ok(Some(Toolchain::new(self, name)?.rustc_version()))
```